scalalib/src/mill/scalalib/GenIdeaImpl.scala

```diff
@@ -148,13 +161,64 @@ object GenIdeaImpl {
       }
       .toMap
 
+    sealed trait ResolvedLibrary { def path : Path }
+    case class CoursierResolved(path : Path, pom : Path, sources : Option[Path])
+      extends ResolvedLibrary
+    case class OtherResolved(path : Path) extends ResolvedLibrary
+
+    // Tries to group jars with their poms and sources.
+    def toResolvedJar(path : Path) : Option[ResolvedLibrary] = {
+      val inCoursierCache = path.startsWith(Path(CoursierPaths.cacheDirectory()))
+      val isSource = path.segments.last.endsWith("sources.jar")
+      val isPom = path.ext == "pom"
+      if (inCoursierCache && (isSource || isPom)) {
+        // Remove sources and pom as they'll be recovered from the jar path
+        None
+      } else if (inCoursierCache && path.ext == "jar") {
+        val withoutExt = path.segments.last.dropRight(path.ext.length + 1)
+        val pom = path / up / s"$withoutExt.pom"
+        val sources = Some(path / up / s"$withoutExt-sources.jar")
+          .filter(_.toIO.exists())
+        Some(CoursierResolved(path, pom, sources))
+      } else Some(OtherResolved(path))
+    }
+
+    // Hack so that Intellij does not complain about unresolved magic
+    // imports in build.sc when in fact they are resolved
+    def sbtLibraryNameFromPom(pom : Path) : String = {
```

Comment:
I should maybe remove this, Intellij seems to be flaky as hell with this. It worked on a java library, doesn't seem to work on scala ones ...

Reply:
Changed to ed958b7. This allows users to avoid squiggly lines by defining their artifacts with ":" rather than "::" (which implies adding `_2.12` to the artifact name)